src/orange/orangeinp/InputBuilder.cc

```diff
@@ -145,6 +182,8 @@ auto InputBuilder::operator()(ProtoInterface const& global) const -> result_type
         csg_outp.write(opts_.csg_output_file);
     }
 
+    convert_logic_to_selected_notation(&result);
```

Comment:
I think this code is in slightly the wrong place: it needs to be in the OrangeParams constructor where we process the OrangeInput, which may come from a converted Proto *or* a JSON file. The `InputBuilder` is only used for the Proto -> Input conversion, not the Input -> Data.

(Actually, how about defining a free public function `void convert_volume_logic(OrangeInput&, LogicNotation)` that is just a null-op if the notation matches or calls the appropriate conversion functions if it doesn't? That would be nice for testing.)

Here since we're always constructing as postfix, the `OrangeInput` built by `InputBuilder` should now *always* be postfix. And OrangeInput built by SCALE will (for the moment) *always* be infix.

Reply:
> Here since we're always constructing as postfix, the `OrangeInput` built by `InputBuilder` should now _always_ be postfix. And OrangeInput built by SCALE will (for the moment) _always_ be infix.

A little bit confused by this sentence and why would SCALE always be infix. `UnitProto`s are first build using postfix notation regardless of what the user provided. The `OrangeInput` should be set to the final desired representation.